R/utilities-checks.R

```diff
@@ -1,3 +1,86 @@
+# Tests -------------------------------------------------------------------
+
+#' Reports whether x is a type of object
+#' @param x An object to test
+#' @keywords internal
+#' @export
+#' @name is_tests
+is.ggplot <- function(x) inherits(x, "ggplot")
+
+#' @export
+#' @rdname is_tests
+is.mapping <- function(x) inherits(x, "uneval")
+
+#' @export
+#' @rdname is_tests
+is.coord <- function(x) inherits(x, "Coord")
+
+#' @export
+#' @rdname is_tests
+#' @usage is.Coord(x) # Deprecated
+is.Coord <- function(x) {
+  deprecate_soft0("3.5.2", "is.Coord()", "is.coord()")
+  is.coord(x)
+}
+
+#' @export
+#' @rdname is_tests
+is.ggproto <- function(x) inherits(x, "ggproto")
+
+#' @export
+#' @rdname is_tests
+is.facet <- function(x) inherits(x, "Facet")
+
+#' @export
+#' @rdname is_tests
+#' @usage
+#' # Layer related tests
+#' is.layer(x)
+is.layer <- function(x) inherits(x, "Layer")
+
+#' @export
+#' @rdname is_tests
+is.geom <- function(x) inherits(x, "Geom")
+
+#' @export
+#' @rdname is_tests
+is.stat <- function(x) inherits(x, "Stat")
+
+#' @export
+#' @rdname is_tests
+is.position <- function(x) inherits(x, "Position")
+
+#' @export
+#' @usage
+#' # Scale and guide related tests
+#' is.scale(x)
+#' @rdname is_tests
+is.scale <- function(x) inherits(x, "Scale")
+
+#' @export
+#' @rdname is_tests
+is.guide <- function(x) inherits(x, "Guide")
+
+#' @export
+#' @rdname is_tests
+is.guides <- function(x) inherits(x, "Guides")
```

Comment:
I don't think this needs exporting as it is an internal structure like Scales

Reply:
`guides()` returns a `<Guides>` class object, so it isn't entirely internal

Reply:
ah, I see - then ignore this